lib/restaurant.rb

```diff
@@ -0,0 +1,49 @@
+class Restaurant
+
+  attr_reader :order_list, :menu, :sms, :calc
+
+  def initialize(args = {})
+    @sms              = args[:sms]              || Messenger.new
+    @menu             = args[:menu]             || Menu.new
+    @calc             = args[:calc]             || Calculator.new
+    @order_list       = []
+  end
+
+  def view_menu
+    menu.display_menu
+  end
```

Comment:
the output for view_menu is quite convoluted after initial list (shows full array of available + unavailable with each dish instance ID as well as the hash for name and price of dish) e.g. #<Dish:0x00007fd9fae8fec8 @name="ramen", @price="12", @available="false">